app/models/configuration_script.rb

```diff
@@ -5,13 +5,23 @@ class ConfigurationScript < ActiveRecord::Base
   include ProviderObjectMixin
 
   def run(vars = {})
-    extra_vars = {'extra_vars' => variables}
-    options = vars.reverse_merge(extra_vars)
+    if vars.present?
+      extra = vars.delete(:extra_vars)
+      extra_vars = merge_extra_vars(extra)
+      options = vars.reverse_merge(extra_vars)
+    end
+
     with_provider_object do |jt|
-      jt.launch(options)
+      jt.launch(options || {})
     end
   end
 
+  def merge_extra_vars(external)
```

Comment:
`external = vars[:extra_var]` can be `nil`, so it fails `internal.merge(external)`